docs/wiki/Cryptography/elliptic-curve-cryptography.md

```diff
@@ -0,0 +1,297 @@
+# A Brief Introduction to Elliptic Curve Cryptography
+
+Cryptography can often seem like an abstract subject, but its applications are deeply woven into the fabric of our everyday lives. Let's consider a real-world scenario to understand how cryptography, specifically Elliptic Curve Cryptography (ECC), can be used to safeguard critical interactions.
+
+Alice, a diligent businesswoman, has been abducted and held captive on a remote island. Her captors demand a hefty ransom of $1 million for her release. With limited options for communication, they provide a single postcard for her to instruct her associate, Bob, to transfer the funds.
+
+Alice considers writing the ransom amount and signing the postcard like a check. However, this method poses a significant risk: the kidnappers could easily forge the postcard, inflate the amount, and deceive Bob into sending them more money.
+
+Alice needs a robust approach that allows:
+
+1. Bob to verify that the transfer has be authorized by her, and
+2. ensure that postcard's message has not been tampered with.
+
+The goal of this exercise is device a method for Alice to create a **secret key 🔑** known only to her. This key will be crucial for her to prove her identity and ensure the message's authenticity to Bob.
+
+Mathematics, as always, comes to the rescue. Through ingenious use of **Elliptic Curves**, let's explore how Alice can generate the **secret key 🔑**.
+
+## Elliptic curves
+
+An elliptic curve is a curve **described by the equation**:
+
+$$
+y^2 = x^3 + ax+b
+$$
+
+Such that $4a^3 + 27b^2 \ne 0$ to ensure the curve is non-singular.
+The equation above is what is called the **Weierstrass normal form** of the long equation:
+
+$$
+y^2 + a_1 xy + a_3 y = x^3 + a_2 x^2 + a_4 x + a_6
+$$
+
+Examples:
+
+<img src="images/elliptic-curves/examples.gif" width="500"/>
+
+Observe that elliptic curves are symmetric about the x-axis.
+
+Ethereum uses a standard curve known as [secp256k1](http://www.secg.org/sec2-v2.pdf) with parameters $a=0$, and $b=7$; which is the curve:
+$$y^2=x^3+7$$
+
+<img src="images/elliptic-curves/secp256k1.png" width="500"/>
+
+## Groups and Fields
+
+### Group
+In mathematics, a **GROUP** is a set $G$, containing at least two elements, which is closed under a binary operation usually referred to as **addition** ($+$). A set is closed under an operation when the result of the operation is also a member of the set. 
```

Comment:
Shouldn't something like this be also included in group definition?   ... Groups have an associative operation and an identity element

Reply:
Ideally it should be. My intent is to make as simple as possible for everyone, regardless of their math background. This definition assumes the reader knows about what inverses while the knowledge of inverse and associative is not applied in the context of ECDSA. Does that sound okay?